backend/src/db/providers/userProvider.ts

```diff
@@ -13,36 +16,52 @@ const getUserProvider = async (userId: string) => {
   return dbUser;
 };
 
+const getFullUserInformationProvider = async (userId: string) => {
+  const dbUserFullInfo = await UserModel.findById(userId)
+    .populate([
+      {
+        path: 'technologies',
+        populate: {
+          path: 'group',
+          model: SkillGroupModel,
+          select: 'name -_id',
+        },
+      },
+      {
+        path: 'technologies',
+        populate: {
+          path: 'achievedSkills',
+          model: UserSkillModel,
+          populate: {
+            path: 'skill',
+            model: SkillModel,
+            select: 'name maxScore image -_id',
+          },
+          select: 'score skill -_id',
+        },
+      },
+    ])
+    .lean();
+
+  return dbUserFullInfo;
+};
+
 const getEmployeesProvider = async (managerId: string) => {
   const employess = await UserModel.find({ managerId }).lean();
   return employess;
 };
 
-const getUserSkills = async (userId: string): Promise<Pick<IUser, 'skills'>> => {
-  const clientSkills = await UserModel.findOne({ _id: userId }, { skills: 1, _id: 0 }).lean();
-  if (!clientSkills) {
-    throw new NotFoundError('User not found.');
-  }
-  return clientSkills;
+const getUserSkills = async (userId: string): Promise<IUserSkill[]> => {
```

Comment:
as Skills it's a  unique entity you can move all skills providers to them own file